### PR TITLE
Remove MLBufferResourceView

### PR DESCRIPTION
**Pull request by a-sully**
Follow-up to #546. `MLBufferResourceView` is not relevant without `MLCommandEncoder`

***
<a href="https://pr-preview.s3.amazonaws.com/a-sully/webnn/pull/553.html" title="Last updated on Feb 8, 2024, 2:41 AM UTC (66588cf)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/webmachinelearning/webnn/553/d1a02f2...a-sully:66588cf.html" title="Last updated on Feb 8, 2024, 2:41 AM UTC (66588cf)">Diff</a>

### CODE REVIEW

File: index.bs
Comment (a-sully) [inline index.bs:1407]: Follow-up question: Do we still need `MLBufferView`?

File: index.bs
Comment (huningxin) [inline index.bs:1408]: Because this PR and #546 remove all WebNN interfaces using `GPUBuffer` and `GPUTexture`, I suppose we should also modify [6.2. Device Selection](https://www.w3.org/TR/webnn/#programming-model-device-selection) section that overviews the usage of `GPUBuffer` and `GPUTexture` for `MLContext` created from a `GPUDevice`.

> In addition to the default method of creation with [MLContextOptions](https://www.w3.org/TR/webnn/#dictdef-mlcontextoptions), an [MLContext](https://www.w3.org/TR/webnn/#mlcontext) could also be created from a specific [GPUDevice](https://www.w3.org/TR/webgpu/#gpudevice) that is already in use by the application, in which case the corresponding [GPUBuffer](https://www.w3.org/TR/webgpu/#gpubuffer) resources used as graph constants, as well as the [GPUTexture](https://www.w3.org/TR/webgpu/#gputexture) as graph inputs must also be created from the same device.

> The following table summarizes the types of resource supported by the context created through different method of creation:

> Creation method | ArrayBufferView | GPUBuffer | GPUTexture
> -- | -- | -- | --
> MLContextOptions | Yes | No | No
> GPUDevice | Yes | Yes | Yes

Comment (a-sully) [reply index.bs:1408]: SGTM. Removed this table in the latest patch

File: index.bs
Comment (bbernhar) [reply index.bs:1397]: > I suppose we won't need it.

I think it depends if `MLBuffer` will be allowed to combine input/output. If so, the WebNN developer could create one large `MLBuffer` to hold inputs and outputs then dispatch them via `MLBufferView`.

File: index.bs
Comment (huningxin) [reply index.bs:1408]: Thanks!

File: index.bs
Comment (huningxin) [reply index.bs:1397]: @a-sully wrote

> I'll put up an issue to discuss

SGTM, thanks!

**Review by huningxin — APPROVED** [review]
LGTM

File: index.bs
Comment (bbernhar) [reply index.bs:1397]: SGTM, @a-sully.

Comment (a-sully) [reply index.bs:1397]: Filed #560. I'll put up a follow-up PR to remove the `MLBufferView` typedef (and another to remove it from the Model Loader API specification, I suppose)

**Comment by bbernhar** [comment]
Thanks, @a-sully LGTM too.

**Comment by a-sully** [comment]
Thank you @bbernhar and @huningxin for the reviews! I have a follow-up queued up to remove `MLBufferView` which I'll put up once this PR merges.

Friendly ping to someone with write access to please merge this PR when you get the chance. Thank you!

**Comment by anssiko** [comment]
Thanks @a-sully for this contribution! Reviewed by @huningxin @bbernhar.

**Pull request merged**
